Support different API version in azure integration (#28524)
backstage/backstage: .changeset/clean-eggs-accept.md

---

> '@backstage/integration': patch

Reviewer: ```suggestion
'@backstage/integration': minor
```

---

> Enabled app-config to take apiVersion of azure integration to support azure devops of different APIs

Reviewer: ```suggestion
Enabled app-config to take `apiVersion` of azure integration to support azure devops of different APIs
```

and then add "apis" (all lowercase) to `.github/vale/config/vocabularies/Backstage/accept.txt` to make the spell checker happy.

Regarding wording, a user reading this in the changelog may want to know that this affects the three functions `getAzureCommitsUrl`, `getAzureDownloadUrl` and `getAzureFileFetchUrl` as well as the type `AzureIntegrationConfig` if they were using them.

> Enabled app-config to take `apiVersion` of azure integration to support azure devops of different APIs

Reviewer 2: ```suggestion
Added support for providing an `apiVersion` for the Azure integration in order to support different Azure DevOps API versions.
```
